product_configurator_wizard/wizard/product_configurator.py

```diff
@@ -121,22 +123,56 @@ def get_form_vals(self, dynamic_fields, domains):
         """
         vals = {}
 
-        dynamic_fields = {k: v for k, v in dynamic_fields.iteritems() if v}
+        dynamic_fields = dynamic_fields.copy()
 
+        # validate and eliminate values, and set defaults if they are on the
+        # current step
+        step_val_ids = cfg_step and \
+            cfg_step.attribute_line_ids.mapped('value_ids').ids or \
+            self.product_tmpl_id.attribute_line_ids.mapped('value_ids').ids
         for k, v in dynamic_fields.iteritems():
+            available_val_ids = domains[k][0][2]
+            # Get this fresh every time as the loop can change the values as
+            # it goes!
+            config_val_ids = [dfv for dfv in dynamic_fields.values()
+                              if dfv and not isinstance(dfv, list)]
+            for list_dfv in [dfv for dfv in dynamic_fields.values()
+                             if dfv and isinstance(dfv, list)]:
+                config_val_ids.extend(list_dfv[0][2])
             if not v:
+                # if the value currently is blank and on the current step, see
+                # if one can be set
+                if set(available_val_ids) & set(step_val_ids):
+                    def_value_id = self.product_tmpl_id.find_default_value(
+                        available_val_ids, config_val_ids
+                    )
+                    if def_value_id:
+                        dynamic_fields.update({k: def_value_id})
+                        vals[k] = def_value_id
                 continue
-            available_val_ids = domains[k][0][2]
             if isinstance(v, list):
                 value_ids = list(set(v[0][2]) & set(available_val_ids))
```

Comment:
`value_ids` could be reduced to nothing, in which case you'd want the default value.